content/epmdlessless.md

```diff
@@ -0,0 +1,124 @@
++++
+author = "Tristan Sloughter"
+type="post"
+categories = ["Erlang"]
+date = 2020-11-30T10:41:00Z
+description = ""
+draft = false
+slug = "epmdlessless"
+tags = ["Erlang"]
+title = "Running Erlang Releases without EPMD on OTP 23.1+"
++++
+
+With Erlang/OTP 23.1 and relx 4.2 (included in rebar3 3.14.3) it is now possible
+to conveniently run Erlang releases without relying on EPMD or any third party
+dependency, like [epmdless](https://github.com/tsloughter/epmdless), for opening a
+remote shell or clustering nodes.
```

Comment:
The sentence fragment with epmdless is a bit odd there. The explanation about EPMD isn't given until later in the text as well. I'm thinking it might make sense to turn this into an intro.

```suggestion
Erlang/OTP deployments that want to provide shell access relied on something
called the Erlang Port Mapper Daemon (EPMD), a separate process that handled
all distribution mechanisms. This often caused issues where local shell access to
a running production system forced the use of peer-to-peer mechanisms that didn't
always work well in all environments. Previous attempts at working around this
requirement used third party dependencies such as
[epmdless](https://github.com/tsloughter/epmdless), but still weren't ideal.

With Erlang/OTP 23.1 and relx 4.2 (included in Rebar3 3.14.3) it is now possible
to conveniently run Erlang releases without relying on EPMD for opening a
remote shell or clustering nodes. This is particularly going to make Erlang releases
simpler to configure in locked down environments, and easier to use with containers.
```

Reply:
Merging this but then changing it a bit.

I'm not sure what you mean by peer-to-peer mechanisms? Is that referring to the remote shell also being a full node listening on a distribution port or to something around how ports were shared between physical nodes in EPMD?

Reply:
yeah. It's not obvious to people sometimes why just getting a REPL involves the same machinery as setting up a mesh cluster.